index.js

```diff
@@ -0,0 +1,179 @@
+
+      // Install the Tutor Pearson Plugin
+      if (tutor_pearson_plugin_url) {
+          core.info('Installing Tutor Pearson plugin');
+          const gh_token_url = tutor_pearson_plugin_url.replace("GH_TOKEN", gh_access_token);
+          await exec.exec('venv/bin/pip', ['install', gh_token_url], options);
+      }
+
+      // Install Tutor plugins
+      if (tutor_plugin_sources) {
+          core.info('Installing Tutor plugins');
+          const plugin_sources = parse_bash_array(tutor_plugin_sources);
+          for (var i=0; i < plugin_sources.length; i++) {
+              let plugin_source = plugin_sources[i];
+              await exec.exec('venv/bin/pip', ['install', plugin_source], options);
+          }
+      }
+
+      // Path to the tutor root will be saved in this variable
+      var tutor_root = "";
+      var tutor_root_options = {
+        shell: '/bin/bash'
+      };
+      tutor_root_options.listeners = {
+        stdout: (data) => {
+          tutor_root += data.toString();
+        },
+        stderr: (data) => {
+          myError += data.toString();
+        }
+      };
+
+      await exec.exec('venv/bin/tutor', ['config', 'save'], options);
+
+      // Running 'tutor config printroot' and saving the output to a variable
+      await exec.exec('venv/bin/tutor', ['config', 'printroot'], tutor_root_options);
+      tutor_root = tutor_root.trim();
+      core.info(`tutor_root: ${tutor_root}`);
+
+      // Create private.txt file
+      await exec.exec('touch', [`${tutor_root}/env/build/openedx/requirements/private.txt`], options);
+
+      // Install extra requirements
+      if (extra_private_requirements) {
+        core.info('Installing extra private requirements');
+        const repositories = parse_bash_array(private_repositories);
+        const branches_array = parse_bash_array(branches);
+        for (var i=0; i< repositories.length; i++) {
+          let repository = repositories[i];
+          let branch = branches_array[i];
+          if (repository == "") {
+            continue;
+          }
+          await exec.exec(
+            'git',
+            [
+              'clone', '-b', branch,
+              `https://${gh_access_token}@github.com/Pearson-Advance/${repository}.git`,
+              `${tutor_root}/env/build/openedx/requirements/${repository}`
+            ],
+            options
+          );
+
+          // Write requirement to the private.txt file
+          fs.appendFileSync(`${tutor_root}/env/build/openedx/requirements/private.txt`, `-e ./${repository}\n`);
+        }
+      }
+
+      // Enable Tutor plugins (global)
+      if (tutor_plugin_names) {
+        core.info('Enabling Tutor plugins');
+        const plugin_names = parse_bash_array(tutor_plugin_names);
+        for (var i=0; i < plugin_names.length; i++) {
+            let plugin_name = plugin_names[i];
+            await exec.exec('venv/bin/tutor', ['plugins', 'enable', plugin_name], options);
+        }
+      }
```

Comment:
Variable used once. @ShonTitor 
```suggestion
        const plugins = parse_bash_array(tutor_plugin_names);
        for (var i=0; i < plugins.length; i++) {
            await exec.exec('venv/bin/tutor', ['plugins', 'enable', plugins[i]], options);
        }
      }
```